OloEngine/src/OloEngine/Scripting/Lua/LuaScriptGlue.cpp

```
@@ -36,20 +38,361 @@
+            // read/write.  Lua scripts never see this pointer directly.
+            [](Entity& e, sol::this_state s) -> sol::object
+            {
+                if (e.HasComponent<T>())
+                    return sol::make_object(s, &e.GetComponent<T>());
+                return sol::make_object(s, sol::nil);
+            }
+        };
+    }
+
+    // clang-format off
+    static const std::unordered_map<std::string_view, ComponentEntry>& GetComponentRegistry()
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)